designs/v1-roadmap.md

```diff
@@ -0,0 +1,270 @@
+
+We need to define a mechanism for communicating breaking changes to the library to projects that rely on it. This library is similar to something like `controller-runtime` , where the library is versioned independent of the projects that rely on it. Starting in v1, we should adopt a versioning scheme similar to this, that decouples the AWS release version from the neutral library version.
+
+#### Tasks
+
+- [ ] Create a design doc for defining a versioning strategy for `kubernetes-sigs/karpenter` and cloud provider repos. Begin adhering to this strategy starting in v1
+
+### NodeClaim Conceptual Documentation
+
+**Issue Ref:** https://github.com/aws/karpenter-provider-aws/issues/5144
+
+**Category:** Stability
+
+Karpenter currently has no conceptual documentation around NodeClaims. NodeClaims have become a fundamental part of how Karpenter launches and manages nodes. There is critical observability information that is stored inside of the NodeClaim that can help users understand when certain disruption conditions are met (Expired, Empty, Drifted) or why the NodeClaim fails to launch.
+
+For Karpenter’s feature completeness at v1, we need to accurately describe to users what the purpose of Karpenter’s NodeClaims are and how to leverage the information that is stored within the NodeClaim to troubleshoot Karpenter’s decision-making.
+
+#### Tasks
+
+- [ ] Add a NodeClaim doc to the “Concepts” section of the documentation
+
+### Drop Knative References from the Code
+
+**Issue Ref:** https://github.com/kubernetes-sigs/karpenter/issues/332
+
+**Category:** Stability
+
+Karpenter has [used knative](https://github.com/knative/pkg) from the beginning of the project. knative’s pkg libraries were only intended for their own use and were not intended to be used widely by the community. Because knative relies on and generates so much of the upstream API, attempting to bump to a newer version of client-go (or any other upstream k8s package) without having knative pkg pinned to that same version causes incompatibilities. Practically, this means that we can be bottlenecked on older versions of k8s libraries while we are waiting on knative to update its own dependencies. [Knative has a slower release cycle than Karpenter](https://github.com/knative/community/blob/main/mechanics/RELEASE-SCHEDULE.md#upcoming-releases) so we need to avoid these bottlenecks while we have the opportunity to make breaking changes to the API.
+
+#### Tasks
+
+- [ ] Remove the knative logger and replace with the controller-runtime logger
+- [ ] Update the status condition schema for Karpenter CustomResources to use the [metav1 status condition schema](https://github.com/kubernetes/apimachinery/blob/f14778da5523847e4c07346e3161a4b4f6c9186e/pkg/apis/meta/v1/types.go#L1523)
+
+### Drop Knative Webhook from Karpenter
```

Comment:
I think we should call out that this is for validation only. 
```suggestion
### Drop Knative Webhook for Validation from Karpenter
```

Reply:
Plan is to drop this long-term generally so I'm fine to have this here since we are definitely going to be fully dropping this in v1.1